src/encrypt/mod.rs

```diff
@@ -0,0 +1,74 @@
+use dusk_bls12_381::Scalar;
+
+use std::io;
+
+/// Encapsulates an encrypted data
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub struct EncryptedData {
+    a: Scalar,
+    b: Scalar,
+}
+
+impl EncryptedData {
+    /// Perform the encryption of a given message provided a secret and a nonce
+    pub fn encrypt(message: &Scalar, secret: &Scalar, nonce: &Scalar) -> Self {
+        let a = *message;
+        let b = secret + nonce;
+
+        Self { a, b }
+    }
+
+    /// Decrypt a previously encrypted message, provided the shared secret and nonce
+    /// used for encryption. If the decryption is not successful, `None` is returned
+    pub fn decrypt(&self, secret: &Scalar, nonce: &Scalar) -> Option<Scalar> {
+        if self.b == secret + nonce {
+            Some(self.a)
+        } else {
+            None
+        }
+    }
+}
+
+impl io::Write for EncryptedData {
+    fn write(&mut self, buf: &[u8]) -> Result<usize, io::Error> {
+        if buf.len() < 64 {
+            return Err(io::Error::from(io::ErrorKind::UnexpectedEof));
+        }
+
+        let mut a = [0x00u8; 32];
+        let mut b = [0x00u8; 32];
+
+        let mut n = a.as_mut().write(&buf[0..32])?;
+        n += b.as_mut().write(&buf[32..64])?;
+
+        // Constant time option is REALLY inflexible, so this is required
+        let a = Scalar::from_bytes(&a);
+        if a.is_some().into() {
+            self.a = a.unwrap();
+        } else {
+            return Err(io::Error::from(io::ErrorKind::InvalidData));
+        }
```

Comment:
I know `CtOption` is not really flexible :/ We might start to think about some utilities / macros for them eventually, in the meantime we might reduce the usage of one branch just having the following:
```rust
if a.is_none().into() {
    return Err(io::Error::from(io::ErrorKind::InvalidData));
}

self.a = a.unwrap();
```

Same for `self.b`.